src/lib/Button/Button.svelte

```diff
@@ -0,0 +1,144 @@
+<!-- @component Button a basic HTML button with Urban styling-->
+<script>
+  import "../style/app.css";
+  import { urbanColors } from "../utils";
+  /**
+   * Which variant of button to use
+   * @type {"primary" | "primary-black" | "secondary" | "secondary-black" | "tertiary"}
+   */
+  export let variant = "primary";
+
+
+  /**
+   * Which size of button to use
+   * @type {"standard" | "small" }
+   */
+  export let size = "standard"
+
+  /**
+   * Is the button disabled?
+   * @type { boolean }
+   */
+  export let disabled = false;
+
+
+  let hovered = false;
+
+  function getIconColor(_variant, _hovered) {
+    // hovered colors
+    if (!_hovered) {
+      if (_variant === "primary" || _variant === "primary-black") {
+        return urbanColors.white;
+      }
+      if (_variant === "secondary") {
+        return urbanColors.blue;
+      }
+      if (_variant === "secondary-black" || _variant === "tertiary") {
+        return urbanColors.black;
+      }
+    }
+    if (_variant === "primary") {
+      return urbanColors.white;
+    }
+    if (_variant === "primary-black") {
+      return urbanColors.black;
+    }
+    // not hovered colors
+    if (_variant === "secondary" || _variant == "secondary-black" || _variant === "tertiary") {
+      return urbanColors.white;
+    }
+    return urbanColors.black;
+  }
+
+  $: iconColor = getIconColor(variant, hovered);
+</script>
+<button class="variant-{variant} size-{size}" on:click {disabled} on:mouseenter={(e) => hovered = true} on:mouseleave={(e) => hovered = false}>
```

Comment:
Should we be adding an event dispatch for click, mouseenter, and mouseleave?

Reply:
yes, good call on mouseenter and mouseleave